src/server/middleware/v2auth/auth.go

```diff
@@ -62,6 +68,9 @@ func (rc *reqChecker) check(req *http.Request) error {
 		}
 	} else if len(middleware.V2CatalogURLRe.FindStringSubmatch(req.URL.Path)) == 1 && !securityCtx.IsSysAdmin() {
 		return fmt.Errorf("unauthorized to list catalog")
+	} else if req.URL.Path == "/v2/" && !securityCtx.IsAuthenticated() {
+		ctx := context.WithValue(req.Context(), middleware.SkipInjectRegistryCredKey, true)
+		*req = *(req.WithContext(ctx))
```

Comment:
We should return 401 and the supported auth method here or in our own handler.

Reply:
Good point, but IMO we may defer this, we can always make sure the registry setting is consistent with the auth setting as the communication between registry and docker or other container runtime is very well tested.

Opened #10610 but it's likely to be deferred to future.